lib/nats.js

```diff
@@ -273,11 +277,17 @@ Client.prototype.selectServer = function() {
   client.url = server.url;
   if ('auth' in server.url && !!server.url.auth) {
     var auth = server.url.auth.split(':');
-    if (client.options.user === undefined) {
-      client.user = auth[0];
-    }
-    if (client.options.pass === undefined) {
-      client.pass = auth[1];
+    if (auth.length !== 1) {
+      if (client.options.user === undefined) {
+        client.user = auth[0];
+      }
+      if (client.options.pass === undefined) {
+        client.pass = auth[1];
+      }
+    } else {
+      if (client.options.token === undefined) {
+        client.token = auth[0];
```

Comment:
I was just pointing out that `client.token` could be be null/empty.  In either case the server will reject as invalid credentials. (no action item).